charts/redpanda/helpers.go

```diff
@@ -62,8 +62,46 @@ func FullLabels(dot *helmette.Dot) map[string]string {
 		"app.kubernetes.io/component":  Name(dot),
 	}
 
-	// TODO Shouldn't our defaults take precedence over user provided labels?
-	return helmette.Merge(defaults, labels)
+	return helmette.Merge(labels, defaults)
```

Comment:
This _could_ be a breaking change and might be worth thinking a bit more about? It would also prevent us from being able to overwrite the chart version in our golden tests, right?

Reply:
> This could be a breaking change and might be worth thinking a bit more about?

Yes, it would be breaking change. My intention was to give user ability to overwrite the defaults. Previously it was named `required`

https://github.com/redpanda-data/helm-charts/blob/bfb4e557ceadfff97a289febbaf5e25fe8b2cdd1/charts/redpanda/templates/_helpers.tpl#L51-L57

> It would also prevent us from being able to overwrite the chart version in our golden tests, right?

Merge works differently. The first argument in the parameters takes precedence. I think MergeOverwrite would work as you think. Currently user can not overwrite defaults map keys. 

https://helm.sh/docs/chart_template_guide/function_list/#mergeoverwrite-mustmergeoverwrite

https://github.com/Masterminds/sprig/blob/581758eb7d96ae4d113649668fa96acc74d46e7f/dict.go#L109-L117

https://github.com/darccio/mergo/blob/cde9f0ea26cccb1168ee3900cf8ca457bb928c3c/merge.go#L59

Reply:
Ooooh, 😵‍💫 Yeah I do not like sprig's argument ordering or examples.